source/RunClient.py

```diff
@@ -35,7 +35,20 @@ def RunClient():
     createDisplay = CreateDisplay(playername)
     handView = HandView(gameControl, createDisplay.display)
     tableView = TableView(createDisplay.display)
+    while(len(tableView.player_names) < 1) or (tableView.player_names.count('guest') > 0 ):
```

Comment:
To avoid duplicate names you need the full list of names (part of  the public info, and found in tableView). 
gameControl only has one player's name.